### ISSUE TITLE
Automatic learning rate for MultiClassOneVsAll

### ISSUE THREAD

**Issue by bduclaux** [catboost/catboost#1198]
Hello

Congrats for the 0.22 release ! The automatic learning rate for MultiClass works great.
Would it be feasible to add the same for MultiClassOneVsAll ?

Also, I was wondering if this could also be available for GPU for both MultiClass and MultiClassOneVsAll. I suspect the algorithm used to determine the optimal learning rate cannot be easily implemented to GPU.

Thanks

**Comment by annaveronika** [comment]
Auto learning rate works for both, cpu and gpu. It is not implemented for MultiClassOneVsAll yet, but we'll add it at some point.

**Comment by RiverShah** [comment]
I am having trouble finding references on how automatic learning rates are being computed. @annaveronika Do you have a paper or explanation on how auto learning rate is computed. In your 2019 London pydata video, you mentioned it is related to number of rows, features and some overfit metric but I could not get a deeper answer from resources. Thank you!

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
The learning rate is set here by default.
https://github.com/catboost/catboost/blob/d534d75702bfa704eea20983788942274585ac44/catboost/libs/train_lib/options_helper.cpp#L247
